pkg/persistence/account/writer/writer.go

```diff
@@ -0,0 +1,169 @@
+/*
+ * @license
+ * Copyright 2023 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package writer
+
+import (
+	"fmt"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/internal/log"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/account"
+	persistence "github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/persistence/account"
+	"github.com/spf13/afero"
+	"gopkg.in/yaml.v2"
+	"path/filepath"
+)
+
+type Context struct {
```

Comment:
Please add documentation to the public functions.
E.g., is OutputFolder already the abs, or the relative one? Depending on that information, L37 could be wrong (as the afero.Fs path is not used).

Reply:
> E.g., is OutputFolder already the abs, or the relative one? Depending on that information, L37 could be wrong (as the afero.Fs path is not used).

100% agree on lacking documentation, but I don't fully understand this question here.

If OutputFolder is relative, it will become absolute, if it is absolute already a `filepath.Abs` does nothing to it.

But what do you mean by:

> as the afero.Fs path is not used

The Fs is used - what do you mean by the "Fs path"?

Reply:
documentation added, but I'm still not sure I understand what concern you have about the "Fs path" - lmk internally if we should discuss this on a quick call

Reply:
>edit: It this afero Fs has a basepath overlay/is a BasePathFs it will append that before the absolute path and create something weird yes (but that's the point of a basepath FS). Is that what you meant?

Yes, that's exactly what I meant. The abs method uses the path of the executable, while the afero basepath is something different. Could lead to some errors depending on the callee code.

Reply:
Ok, then I got what you mean. 

However: We directly use filepath.Abs in several places that deal with files already - and afero's basepath Fs in very few other cases (convert, manifest load and the new account E2E test). 
(46 vs 5 usage search hits)

I think this is well worth discussing - e.g. we might want to just build fitting basepath filesystems at the very top of commands and not care about relative or absolute paths in most code. 
But for this piece of code I would for now lean towards keeping the direct use of filepath.Abs like most other places loading files, even though it might cause issues for basepath filesystems. 

IMO we should